src/convolution_api.cpp

```diff
@@ -295,105 +295,140 @@ miopenConvolutionForwardGetWorkSpaceSize(miopenHandle_t handle,
-                << " --conv_stride_d " << miopen::deref(convDesc).GetConvStrides()[0]
-                << " -u " << miopen::deref(convDesc).GetConvStrides()[1]
-                << " -v " << miopen::deref(convDesc).GetConvStrides()[2]
-                << " --dilation_d " << miopen::deref(convDesc).GetConvDilations()[0]
-                << " -l " << miopen::deref(convDesc).GetConvDilations()[1]
-                << " -j " << miopen::deref(convDesc).GetConvDilations()[2]
-                << " --spatial_dim 3"; // clang-format on
-            std::string x_layout = miopen::deref(xDesc).GetLayout("NCDHW");
-            std::string w_layout = miopen::deref(wDesc).GetLayout("NCDHW");
-            std::string y_layout = miopen::deref(yDesc).GetLayout("NCDHW");
-            if(x_layout != "NCDHW")
-                ss << " --in_layout " << x_layout;
-            if(w_layout != "NCDHW")
-                ss << " --fil_layout " << w_layout;
-            if(y_layout != "NCDHW")
-                ss << " --out_layout " << y_layout;
+            ss << " --out_layout " << y_layout;
         }
-        ss << " -m " << (miopen::deref(convDesc).mode == 1 ? "trans" : "conv") // clang-format off
-            << " -g " << miopen::deref(convDesc).group_count
-            << " -F " << std::to_string(static_cast<int>(conv_dir))
-            << " -t 1"; // clang-format on
-        if(miopen::deref(xDesc).GetType() == miopenInt8x4)
-            ss << " -Z 1";
-        if(is_immediate)
-            ss << " -S 0";
-        MIOPEN_LOG_DRIVER_CMD(ss.str());
+    }
+    ss << " -m " << (miopen::deref(convDesc).mode == 1 ? "trans" : "conv") // clang-format off
+        << " -g " << miopen::deref(convDesc).group_count
+        << " -F " << std::to_string(static_cast<int>(conv_dir))
+        << " -t 1"; // clang-format on
+    if(miopen::deref(xDesc).GetType() == miopenInt8x4)
+        ss << " -Z 1";
+    if(is_immediate)
+    {
+        ss << " -S 0";
+    }
+
+    return ss.str();
+}
+
+void LogCmdConvolution(const miopenTensorDescriptor_t& xDesc,
+                       const miopenTensorDescriptor_t& wDesc,
+                       const miopenConvolutionDescriptor_t& convDesc,
+                       const miopenTensorDescriptor_t& yDesc,
+                       const miopen::ConvDirection& conv_dir,
+                       bool is_immediate)
+{
+    if(miopen::IsLoggingCmd())
+    {
+        const std::string& str =
+            ConvArgsForMIOpenDriver(xDesc, wDesc, convDesc, yDesc, conv_dir, is_immediate);
+        MIOPEN_LOG_DRIVER_CMD(str);
+    }
+}
+
+void LogCmdFindConvolution(const miopenTensorDescriptor_t& xDesc,
+                           const miopenTensorDescriptor_t& wDesc,
+                           const miopenConvolutionDescriptor_t& convDesc,
+                           const miopenTensorDescriptor_t& yDesc,
+                           const miopen::ConvDirection& conv_dir,
+                           bool is_immediate)
+{
+    if(miopen::IsLoggingCmdFind())
```

Comment:
We do not need separate env variable because it is easy to distinguish FindConvolution from other conv calls looking at function name. This is output of FindConvolution:
```
MIOpen(HIP): Command [LogCmdFindConvolution] ./bin/MIOpenDriver ...
```
This is output of other convolution calls:
```
MIOpen(HIP): Command [LogCmdConvolution] ./bin/MIOpenDriver ...
```
You can `grep -v LogCmdFindConvolution` to exclude FindConvolution calls from the log or you can `grep -v LogCmdConvolution` to exclude other convolution calls.

Reply:
@JehandadKhan had some input regarding this [here ](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1662#discussion_r947029795).

Reply:
But I do agree with you. We just have to mindful while grepping.